tmlibrary/tmlib/tools/classification.py

```diff
@@ -114,3 +169,83 @@ def process_request(self, submission_id, payload):
             self.save_result_values(
                 mapobject_type_name, result_id, predicted_labels
             )
+
+    def save_selections(self, submission_id, payload):
+        '''Processes a client tool request and inserts the generated result
+        into the database. This function deals with the save labels jobs.
+        The `payload` is expected to have the following form::
+
+            {
+                "choosen_object_type": str,
+                "selected_features": [str, ...],
+                "training_classes": [
+                    {
+                        "name": str,
+                        "object_ids": [int, ...],
+                        "color": str
+                    },
+                    ...
+                ],
+                "options": {
+                    "method": str,
+                    "n_fold_cv": int
+                },
+                "task": str (either 'classification' or 'saveLabels'),
+                name: str
+
+            }
+
+        Parameters
+        ----------
+        submission_id: int
+            ID of the corresponding job submission
+        payload: dict
+            description of the tool job
+        '''
+        logger.info('Save current selections')
+        mapobject_type_name = payload['chosen_object_type']
+
+        labels = dict()
+        label_map = dict()
+        for i, cls in enumerate(payload['training_classes']):
+            labels.update({j: float(i) for j in cls['object_ids']})
+            label_map[float(i)] = {'name': cls['name'], 'color': cls['color']}
```

Comment:
```suggestion
        for cls_id, cls in enumerate(payload['training_classes']):
            labels.update({val: float(cls_id) for val in cls['object_ids']})
            label_map[float(cls_id)] = {'name': cls['name'], 'color': cls['color']}
```

I'd avoid single-letter variables as much as I can.

Question: are you sure that there are no "overlapping" values in `cls['objects_id']` that could end up as the same key of those dictionaries?

Reply:
@bebosudo This isn't actually new code, but just a copy of how the classifier extracted the object_ids from the payload. Given that it worked robustly there, I assume that's not an issue, but I don't have a good setup to test that